front/src/applications/operationalStudies/components/MacroEditor/export.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  osrdEditoastApi,
+  type Distribution,
+  type TrainScheduleBase,
+} from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+import type { NGEDeleteEvent, NGEEvent, NGEUpdateEvent, TrainrunSectionOperation } from './types';
+
+const defautlPayloads = {
+  constraint_distribution: 'STANDARD' as Distribution,
+  rolling_stock_name: '',
+  start_time: '2024-07-15T08:00:00+02:00',
+};
+
+const ngeCreateOperation = (trainrunSection: TrainrunSectionOperation) => {
+  const { sourceNode, targetNode } = trainrunSection;
+  const [sourceTrigram, sourceSecondaryCode] = sourceNode.betriebspunktName.split('/');
+  const [targetTrigram, targetSecondaryCode] = targetNode.betriebspunktName.split('/');
+  const payloads: TrainScheduleBase[] = [
+    {
+      ...defautlPayloads,
+      path: [
+        {
+          trigram: sourceTrigram,
+          secondary_code: sourceSecondaryCode,
+          id: `${sourceNode.id}`,
+          deleted: false,
+        },
+        {
+          trigram: targetTrigram,
+          secondary_code: targetSecondaryCode,
+          id: `${targetNode.id}`,
+          deleted: false,
+        },
+      ],
+      train_name: trainrunSection.trainrun.name,
+    },
+  ];
+  return payloads;
+};
+
+const ngeUpdateOperation = (trainrunSections: TrainrunSectionOperation[]) => {
+  const path = trainrunSections.flatMap((section, index) => {
+    const [sourceTrigram, sourceSecondaryCode] = section.sourceNode.betriebspunktName.split('/');
+    const [targetTrigram, targetSecondaryCode] = section.targetNode.betriebspunktName.split('/');
+    const sourceNode = {
+      trigram: sourceTrigram,
+      secondary_code: sourceSecondaryCode,
+      id: `${section.sourceNode.id}-${index}`,
+    };
+    const targetNode = {
+      trigram: targetTrigram,
+      secondary_code: targetSecondaryCode,
+      id: `${section.targetNode.id}-${index}`,
+    };
+    if (index === trainrunSections.length - 1) {
+      return [sourceNode, targetNode];
+    }
+    return [sourceNode];
+  });
+  const [firstTrainrunSection] = trainrunSections;
+  const payloads: TrainScheduleBase = {
+    ...defautlPayloads,
+    path,
+    train_name: firstTrainrunSection.trainrun.name,
+  };
+  return payloads;
+};
+
+const ngeOperations = async (event: NGEEvent, dispatch: AppDispatch, timeTableId: number) => {
+  switch (event.type) {
+    case 'create': {
+      dispatch(
+        osrdEditoastApi.endpoints.postV2TimetableByIdTrainSchedule.initiate({
+          id: timeTableId!,
```

Comment:
Nit: `!` should be unnecessary here because `timeTableId` isn't nullable

Reply:
Done